### DISCUSSION TITLE
[Desglose][WIP] Oleada de enemigos

### DISCUSSION THREAD

**Discussion by emrodk** [Godot-en-Espanol/Node-City#14] [Tareas]
# War Director WIP (Work In Progress)
En el juego original cada stage consistía en un pool de 20 tanques. Comenzaba con 4 tanques y cada vez que un tanque era destruido se reponía hasta obtener 4 tanques en pantalla, así hasta que no quedara ninguno. 

#Propuesta Node City
En Node City vamos a cambiar la mecánica orignial por oleadas de enemigos. Es decir que cada stage consistirá de un número de olas, donde cada ola consiste en un grupo de sinergia de enemigos. Una vez que el jugador vence una ola se pasa a la siguiente, así hasta vencer todas las olas.

## Grupos de sinergia (Probablemente se lo trabaje como un submodulo del Director)
Se trata de armar grupos de enemigos, combinando aquellos tipos de tanques que por su manera de comportarse presentan una problemática única para el jugador, el cual deberá plantearse estrategias para lidiar con la misma. De esta manera es posible definir diferentes dificultades.

## Requerimientos
- Poder setear la cantidad de olas por stage
- El número máximo de enemigos que puede tener una ola
- Setear reglas que permitan seleccionar grupos de sinergia, esto es clave para poder tener un control sobre la dificultad del stage. Por ejemplo, si se spawnea un grupo de sinergia muy difícil tras otro podría llegar a abrumar a un jugador que recién comienza.
- Entre ola y ola hay un "peace time" (en segundos), debería poder setearse la duración en el inspector.

## Señales
* Emite
  * Cuando la ola es destruida
    * Si no es la última ola comienza el "peace time" y una vez finalizado arranca la siguiente
* Recibe
  * Cuando un tanque es destruido.
    * Si es enemigo registra la baja. Esto será utilizado para contabilizar el score.
    * Si es el player, detiene el sistema
    *

**Comment by LeandroCuello** [comment]
Lo único que pienso es tal vez el la señal mande la variable de cuantas rondas falta para que se actualiza la información en el HUD. Aunque no se si hablamos de que se muestre la cantidad de rondas que falten.

**Reply by FROSADO** [reply]
Creo que la señal de "fin de oleada" es el que avisa al contador de oleadas, que es el que lleva la cuenta de cuantas quedan. 
Una vez terminadas todas las oleadas, se lanza un evento de "fin de oleadas", que indica el final del .. ¿nivel?
